backend/core/domains/payments/serializers.py

```
@@ -257,12 +564,210 @@ class Meta:
+        """Validate payment request data"""
+        import logging
+        logger = logging.getLogger(__name__)
+
+        logger.info(f"🔍 PAYMENT VALIDATION - Received data: {data}")
+
+        # Validate custom amount if payment_type is CUSTOM
+        if data.get('payment_type') == 'CUSTOM':
+            if not data.get('amount'):
+                raise serializers.ValidationError(
+                    "amount is required when payment_type is CUSTOM"
+                )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)